### PR TITLE
migrate to indy apm-error-logging-plugin

### LINKED ISSUE #1337: Migrate plugins to indy plugins
A meta-issue that tracks the migration

- [x] apm-apache-httpclient-plugin https://github.com/elastic/apm-agent-java/pull/1349
- [x] apm-api-plugin https://github.com/elastic/apm-agent-java/pull/1411
- [x] apm-asynchttpclient-plugin elastic/apm-agent-java#1351
- [x] apm-bootdelegation-plugin N/A - might be removed once all plugins have been migrated to indy plugins
- [x] apm-dubbo-plugin https://github.com/elastic/apm-agent-java/pull/2087
- [x] apm-error-logging-plugin https://github.com/elastic/apm-agent-java/pull/1362
- [x] apm-es-restclient-plugin #2088
- [x] apm-grails-plugin https://github.com/elastic/apm-agent-java/pull/1371
- [x] apm-grpc https://github.com/elastic/apm-agent-java/issues/1358
- [x] apm-hibernate-search-plugin https://github.com/elastic/apm-agent-java/pull/1404
- [x] apm-httpclient11-plugin
- [x] apm-httpclient-core n/a
- [x] apm-java-concurrent-plugin
- [x] apm-jaxrs-plugin https://github.com/elastic/apm-agent-java/pull/1374
- [x] apm-jaxws-plugin elastic/apm-agent-java#1383
- [x] apm-jdbc-plugin
- [x] apm-jms-plugin https://github.com/elastic/apm-agent-java/pull/1394
- [x] apm-jmx-plugin N/A
- [x] apm-jsf-plugin https://github.com/elastic/apm-agent-java/pull/1405
- [x] apm-kafka-plugin #2095 
- [x] apm-log-correlation-plugin N/A
- [x] apm-log-shipper-plugin N/A
- [x] apm-micrometer-plugin
- [x] apm-mongoclient-plugin https://github.com/elastic/apm-agent-java/pull/1369
- [x] apm-okhttp-plugin https://github.com/elastic/apm-agent-java/pull/1368
- [x] apm-opentracing-plugin #1410
- [x] apm-process-plugin
- [x] apm-profiling-plugin N/A
- [x] apm-quartz-job-plugin https://github.com/elastic/apm-agent-java/pull/1366
- [x] apm-redis-plugin #2090
- [x] apm-scala-concurrent-plugin https://github.com/elastic/apm-agent-java/pull/1367
- [x] apm-scheduled-annotation-plugin https://github.com/elastic/apm-agent-java/pull/1364
- [x] apm-servlet-plugin
- [x] apm-spring-resttemplate-plugin https://github.com/elastic/apm-agent-java/pull/1363
- [x] apm-spring-webmvc-plugin https://github.com/elastic/apm-agent-java/pull/1365
- [x] apm-urlconnection-plugin
- [x] trace_methods instrumentation #2094 

Prerequisite for elastic/apm-agent-java#1338 and elastic/apm-agent-java#1339

### PR DESCRIPTION
**Pull request by kananindzya**
refer #1337

### CODE REVIEW

File: apm-agent-plugins/apm-error-logging-plugin/src/main/java/co/elastic/apm/agent/errorlogging/AbstractLoggerErrorCapturingInstrumentation.java [hunk 1]
```diff
@@ -42,49 +42,39 @@
...
 
-    @SuppressWarnings({"WeakerAccess"})
-    @VisibleForAdvice
-    public static final ThreadLocal<Boolean> nestedThreadLocal = new ThreadLocal<Boolean>() {
-        @Override
-        protected Boolean initialValue() {
-            return Boolean.FALSE;
-        }
-    };
+    public static final GlobalThreadLocal<Boolean> nestedThreadLocal = GlobalThreadLocal.get(AbstractLoggerErrorCapturingInstrumentation.class, "nested", Boolean.FALSE);
```
Comment (felixbarny) [inline apm-agent-plugins/apm-error-logging-plugin/src/main/java/co/elastic/apm/agent/errorlogging/AbstractLoggerErrorCapturingInstrumentation.java:45]: Can be simplified with `co.elastic.apm.agent.sdk.state.CallDepth`. See `co.elastic.apm.agent.servlet.RequestStreamRecordingInstrumentation.GetInputStreamAdvice` as an example.

When using it, make sure to add `onThrowable = Throwable.class` to `OnMethodExit`. See also the Javadocs of `co.elastic.apm.agent.sdk.state.CallDepth#decrement` for more details.

Comment (kananindzya) [reply apm-agent-plugins/apm-error-logging-plugin/src/main/java/co/elastic/apm/agent/errorlogging/AbstractLoggerErrorCapturingInstrumentation.java]: replaced with `CallDepth`

File: apm-agent-plugins/apm-error-logging-plugin/src/main/java/co/elastic/apm/agent/errorlogging/AbstractLoggerErrorCapturingInstrumentation.java [hunk 2]
```diff
@@ -42,50 +42,37 @@
...
-        @Advice.OnMethodExit(suppress = Throwable.class)
-        public static void logExit(@Advice.Local("nested") boolean nested,
-                                   @Advice.Local("error") @Nullable ErrorCapture error) {
-            if (error != null) {
-                error.deactivate().end();
+        @Advice.OnMethodExit(suppress = Throwable.class, onThrowable = Throwable.class, inline = false)
+        public static void logExit(@Advice.Enter @Nullable Object errorCaptureObj) {
+            if (callDepth.isNestedCallAndDecrement()) {
+                return;
             }
```
Comment (felixbarny) [inline apm-agent-plugins/apm-error-logging-plugin/src/main/java/co/elastic/apm/agent/errorlogging/AbstractLoggerErrorCapturingInstrumentation.java:72]: Can be simplified as there's already a `if (errorCaptureObj instanceof ErrorCapture)` check afterwards. That check is only true if the error object is not null.

```suggestion
            callDepth.decrement();
```

Comment (kananindzya) [reply apm-agent-plugins/apm-error-logging-plugin/src/main/java/co/elastic/apm/agent/errorlogging/AbstractLoggerErrorCapturingInstrumentation.java]: simplified

**Review by felixbarny — APPROVED** [review]

**Review by SylvainJuge — APPROVED** [review]

**Pull request merged**